opus/import/populate_obs_instrument_VGUVS_prof.py

```diff
@@ -0,0 +1,611 @@
+    if _is_Voyager_at_north(**kwargs):
+        inc = populate_obs_ring_geometry_VGUVS_incidence2_PROF(**kwargs)
+        return inc - 90.
+    else:
+        inc = populate_obs_ring_geometry_VGUVS_incidence1_PROF(**kwargs)
+        return 90. - inc
+
+# Ring elevation to solar, same to opening angle except, it's positive if
+# source is at north side of Jupiter, Saturn, and Neptune, and south side of
+# Uranus. Negative if source is at south side of Jupiter, Saturn, and Neptune,
+# and north side of Uranus.
+def populate_obs_ring_geometry_VGUVS_solar_ring_elevation1_PROF(**kwargs):
+    if _is_Voyager_at_north(**kwargs):
+        target_name = populate_target_name_from_index(**kwargs)
+        if target_name =='U RINGS':
+            inc = populate_obs_ring_geometry_VGUVS_incidence2_PROF(**kwargs)
+            el = 90. - inc # positive
+        else:
+            inc = populate_obs_ring_geometry_VGUVS_incidence1_PROF(**kwargs)
+            el = inc - 90. # negative
+    else:
+        inc = populate_obs_ring_geometry_VGUVS_incidence2_PROF(**kwargs)
+        el = 90. - inc # positive
+
+    return el
+
+def populate_obs_ring_geometry_VGUVS_solar_ring_elevation2_PROF(**kwargs):
+    if _is_Voyager_at_north(**kwargs):
+        target_name = populate_target_name_from_index(**kwargs)
+        if target_name =='U RINGS':
+            inc = populate_obs_ring_geometry_VGUVS_incidence1_PROF(**kwargs)
+            el = 90. - inc # positive
+        else:
+            inc = populate_obs_ring_geometry_VGUVS_incidence2_PROF(**kwargs)
+            el = inc - 90. # negative
+    else:
+        inc = populate_obs_ring_geometry_VGUVS_incidence1_PROF(**kwargs)
+        el = 90. - inc # positive
+
+    return el
+
+def populate_obs_ring_geometry_VGUVS_ring_intercept_time1_PROF(**kwargs):
+    return populate_time1_from_index(column='RING_EVENT_START_TIME', **kwargs)
+
+def populate_obs_ring_geometry_VGUVS_ring_intercept_time2_PROF(**kwargs):
+    return populate_time1_from_index(column='RING_EVENT_STOP_TIME', **kwargs)
+
+def _is_Voyager_at_north(**kwargs):
+    """In this volume,
+    Voyager is at north when:
+        - Source is DELTA SCO (Target S RINGS)
+        - Source is SIGMA SGR (Target U RINGS, start time before
+          1986-01-24T17:10:13.320)
+    Voyager is south when:
+        - Source is SIGMA SGR (Target N RINGS, start time after
+          1986-01-24T17:10:13.320)
+        - Source is IOTA HER (Target S RINGS)
+    """
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    src_name = index_row['SIGNAL_SOURCE_NAME_1']
+    target_name = index_row['TARGET_NAME'].upper().strip()
+    if (src_name == "DELTA SCO"
+        or (src_name == "SIGMA SGR" and target_name == "U RINGS")):
+        return True
+    else:
+        return False
+
+
+################################################################################
+# THESE NEED TO BE IMPLEMENTED FOR EVERY voyager INSTRUMENT
```

Comment:
Another bad search-and-replace that might be historical but you can fix. Make VOYAGER all caps in this comment in all relevant files.